### ISSUE TITLE
Delay when deleting files.

### ISSUE THREAD

**Issue by AleMello** [files-community/Files#14297]
### Description

Everytime I delete one or more files it take a long time. The app becomes unresponsible, and if I try to click anywhere, the window title bar shows (not responding)

https://github.com/files-community/Files/assets/4070249/b5346690-efe9-4cc6-8fe1-a1c4874d6bd6

### Steps To Reproduce

Just click a fie or more and hit delete (both in menu or keyboard)

### Requirements

- e.g. Fix crash that occurs when I delete files

### Files Version

3.0.15.0

### Windows Version

11 pro 23H2 build 22631.2861

### Log File

[debug.log](https://github.com/files-community/Files/files/13773309/debug.log)

**Comment by yaira2** [comment]
Are there a lot of items already in Recycle Bin?

**Comment by AleMello** [comment]
> Are there a lot of items already in Recycle Bin?

1591. I recorded cleaning the Recycle Bin and tried again...

It took a lot of time (8 min 30 sec) to finish the cleaning. File Explorer shown the empty bin (shortly after), but Files was struggling.

After this, deleting files was fast as it should be. Anyway, I think it's an attention point, since I don't wait so long to delete files when using File Explorer (no matter how many files are in the bin). Sometimes I open file explorer just to delete files and go back to Files to work.

I have 4 disks in my computer (as seen in the mp4 I attached). One of it is a HDD. The other 3 are SSD (one of them is a M2, where windows is). I was considering if it would be a problem to Files, since HDD is much slower (720rpm)...

https://github.com/files-community/Files/assets/4070249/0a289f53-9ad8-4cd8-82c5-e3e39b93c9f5

Thanks for your help! I love Files app, and recommend it to everyone I know.
